### PR TITLE
allow passing options to react hooks module

### LINKED ISSUE #92: Opening the API to allow for custom React Context
## Context
- `react-redux` currently supports providing a custom context [during Provider creation](https://react-redux.js.org/api/provider#props) via the the `context` prop as well as in [its `createXXXHook` utilities](https://github.com/reduxjs/react-redux/blob/master/src/hooks/useSelector.js#L93);
- `rtk-query` hooks will [reach out to the default `ReactReduxContext` context](https://github.com/rtk-incubator/rtk-query/blob/next/src/buildHooks.ts#L3) with currently no option to override;
- This makes it so consumers of the `context` api can't use `rtk-query` in their existing projects (ohno).

## Question
Do you consider adding some kind of option to the `createAPI` API that would allow consumers to provide a custom context which would by the generated hooks to reach out to the store?

I'd be glad to contribute, but wanted to run the question through you first, as I imagine you might have some opinions on this topic.

### PR DESCRIPTION
**Pull request by phryneas**
Closes #92

### CODE REVIEW

File: docs/concepts/customizing-create-api.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+---
+
+# Customizing `createApi`
+
+Right now, we are shipping `createApi` in two flavors:
+
+- `createPureApi` which contains only the basic redux logic (the core module)
+- `createApi` which contains the core module and additionally the react-hooks module
+
+You can create your own versions of `createApi` by either specifying non-default options for the modules, or add your own modules.
```
Comment (msutkowski) [inline docs/concepts/customizing-create-api.md:15]:
```suggestion
You can create your own versions of `createApi` by either specifying non-default options for the modules or by adding your own modules.
```

File: docs/concepts/customizing-create-api.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+---
+id: customizing-create-api
+title: Customizing createApi
+sidebar_label: Customizing createApi
+hide_title: true
+---
+
+# Customizing `createApi`
+
+Right now, we are shipping `createApi` in two flavors:
```
Comment (msutkowski) [inline docs/concepts/customizing-create-api.md:10]:
```suggestion
Currently, RTK Query includes two variants of `createApi`:
```

File: docs/concepts/customizing-create-api.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+- `createPureApi` which contains only the basic redux logic (the core module)
+- `createApi` which contains the core module and additionally the react-hooks module
```
Comment (msutkowski) [inline docs/concepts/customizing-create-api.md:13]:
```suggestion
- `createApi` which contains both the core and react-hooks modules
```

File: docs/concepts/customizing-create-api.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import { createDispatchHook, ReactReduxContextValue } from 'react-redux';
+import { buildCreateApi, coreModule, reactHooksModule } from '@rtk-incubator/rtk-query';
+
+const MyContext = React.createContext<ReactReduxContextValue>(null as any);
+const customCreateApi = buildCreateApi(coreModule(), reactHooksModule({ useDispatch: createDispatchHook(MyContext) }));
+```
+
+## Creating your own module
+
+If you want to create your own module, best take a lookt at [the react-hooks module](https://github.com/rtk-incubator/rtk-query/blob/next/src/react-hooks/module.ts) as what such an implementation would look like.
```
Comment (msutkowski) [inline docs/concepts/customizing-create-api.md:33]:
```suggestion
If you want to create your own module, you should review [the react-hooks module](https://github.com/rtk-incubator/rtk-query/blob/next/src/react-hooks/module.ts) to see what an implementation would look like.
```

File: docs/concepts/customizing-create-api.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import { CoreModule } from '@internal/core/module';
+import { BaseQueryFn, EndpointDefinitions, Api, Module, buildCreateApi, coreModule } from '@rtk-incubator/rtk-query';
+
+export const customModuleName = Symbol();
+export type CustomModule = typeof customModuleName;
+
+declare module '@rtk-incubator/rtk-query' {
+  export interface ApiModules<
+    BaseQuery extends BaseQueryFn,
+    Definitions extends If you have created a moduleEndpointDefinitions,
```
Comment (msutkowski) [inline docs/concepts/customizing-create-api.md:47]:
```suggestion
    Definitions extends EndpointDefinitions,
```

File: docs/concepts/customizing-create-api.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+- `createPureApi` which contains only the basic redux logic (the core module)
+- `createApi` which contains the core module and additionally the react-hooks module
+
+You can create your own versions of `createApi` by either specifying non-default options for the modules, or add your own modules.
+
+## Customizing the React-Redux Hooks
+
+If you want the hooks to use different versions of `useSelector` or `useDispatch`, for example if you are using a custom context, you can pass these in at module creation:
+If you have created a module
```
Comment (msutkowski) [inline docs/concepts/customizing-create-api.md:20]:
```suggestion
```

**Review by msutkowski — CHANGES_REQUESTED** [review]
Let's make some modules!

File: docs/concepts/customizing-create-api.md [hunk 7]
```diff
@@ -30,7 +29,7 @@ const customCreateApi = buildCreateApi(coreModule(), reactHooksModule({ useDispa
 
 ## Creating your own module
 
-If you want to create your own module, best take a lookt at [the react-hooks module](https://github.com/rtk-incubator/rtk-query/blob/next/src/react-hooks/module.ts) as what such an implementation would look like.
+If you want to create your own module, you should review [the react-hooks module](https://github.com/rtk-incubator/rtk-query/blob/next/src/react-hooks/module.ts) to see what an implementation would look like.
```
Comment (markerikson) [inline docs/concepts/customizing-create-api.md:32]: :hand: should the link to go `main` instead of `next` ?

Comment (phryneas) [reply docs/concepts/customizing-create-api.md:32]: Hmm, good question. Yeah, for documentation purposes `main` is probably better.

File: docs/concepts/customizing-create-api.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+import { createDispatchHook, ReactReduxContextValue } from 'react-redux';
+import { buildCreateApi, coreModule, reactHooksModule } from '@rtk-incubator/rtk-query';
+
+const MyContext = React.createContext<ReactReduxContextValue>(null as any);
+const customCreateApi = buildCreateApi(coreModule(), reactHooksModule({ useDispatch: createDispatchHook(MyContext) }));
+```
+
+## Creating your own module
+
+If you want to create your own module, you should review [the react-hooks module](https://github.com/rtk-incubator/rtk-query/blob/next/src/react-hooks/module.ts) to see what an implementation would look like.
```
Comment (phryneas) [inline docs/concepts/customizing-create-api.md:32]:
```suggestion
If you want to create your own module, you should review [the react-hooks module](https://github.com/rtk-incubator/rtk-query/blob/main/src/react-hooks/module.ts) to see what an implementation would look like.
```

**Review by msutkowski — APPROVED** [review]

**Pull request merged**